### ISSUE TITLE
tfp.bijectors.IteratedSigmoidCentered gives negative probabilities in inverse mode

### ISSUE THREAD

**Issue by nthakor** [tensorflow/probability#411]
tfp.bijectors.IteratedSigmoidCentered simulates the stick breaking process. That means while using in forward or backward the output must be positive as they are probabilities. 
When I run following command
```
pi = tf.random.uniform([20,10])
pi = tf.nn.softmax(pi)
vi = tfp.bijectors.IteratedSigmoidCentered().inverse(pi)
vi<0
```

Output is 

```
<tf.Tensor: id=6021434, shape=(20, 9), dtype=bool, numpy=
array([[False,  True, False, False,  True,  True, False,  True, False],
       [False, False,  True, False, False,  True,  True, False, False],
       [ True,  True, False, False, False, False,  True, False, False],
       [ True,  True, False, False, False,  True, False,  True, False],
       [ True,  True,  True,  True, False, False, False,  True,  True],
       [False, False, False, False,  True, False, False,  True, False],
       [ True, False,  True,  True,  True, False, False, False, False],
       [False,  True,  True, False,  True,  True,  True, False,  True],
       [False,  True,  True,  True,  True, False,  True,  True, False],
       [False, False,  True, False, False,  True,  True, False,  True],
       [ True,  True,  True,  True, False, False, False, False, False],
       [False,  True, False,  True,  True,  True, False, False,  True],
       [ True, False,  True, False, False,  True,  True,  True, False],
       [ True,  True,  True, False,  True,  True,  True,  True, False],
       [ True,  True,  True,  True, False,  True,  True, False, False],
       [ True, False, False,  True,  True, False,  True, False,  True],
       [False,  True,  True,  True, False,  True,  True, False, False],
       [ True, False, False,  True, False, False,  True, False, False],
       [False, False, False, False,  True,  True,  True,  True,  True],
       [ True, False,  True, False, False, False, False, False,  True]])>
```

**Comment by nthakor** [comment]
tfp.bijectors.IteratedSigmoidCentered provides

**Comment by SiegeLordEx** [comment]
Thanks for the report. It's true that semantically both image and domain of this bijector are probabilities. However, in terms of its implementation, the domain of this bijector is actually encoded as logits in R^N. This in fact is the reason why this bijector is useful, it lets you map an unconstrained space (which is easy to sample from for MCMC and gradient based optimization like variational inerence) to the constrained space of probabilities (which might be useful for specifying certain models).

To convert the output of the inverse of this bijector to probabilities, you can do something along the lines of this: https://github.com/tensorflow/probability/blob/master/tensorflow_probability/python/bijectors/iterated_sigmoid_centered.py#L117-L121 Note that the first line computes an offset such that if you pass it a K-vector of zeros, that corresponds to a K vector of probabilities where each element is 1/K.
